src/main/scala/is/hail/io/bgen/LoadBgen.scala

```diff
@@ -109,15 +109,30 @@ object LoadBgen {
       val rv = RegionValue(region)
 
       it.map { case (_, record) =>
-        val v = record.getKey
-        val vRecoded = v.copy(contig = contigRecoding.getOrElse(v.contig, v.contig))
-        gr.checkVariant(vRecoded)
+        val (contig, pos, alleles) = record.getKey
+        val contigRecoded = contigRecoding.getOrElse(contig, contig)
 
         region.clear()
         rvb.start(kType)
         rvb.startStruct()
-        rvb.addAnnotation(kType.fieldType(0), vRecoded.locus) // locus/pk
-        rvb.addAnnotation(kType.fieldType(1), IndexedSeq(vRecoded.ref, vRecoded.alt))
+
+        gr match {
+          case Some(gr) =>
```

Comment:
what is the purpose of constructing the locus here? is it to check that it's valid for the given genome reference?

Reply:
I think maybe this could be `rvb.addAnnotation(kType.fieldType(0), Locus.fromGR(contig, pos, gr))` if you define `Locus.fromGR` as above. This wraps the gr = None case in a Row which you're not currently doing, but looks slightly tidier? I don't feel as strongly about this, either way.